web/views.py

```diff
@@ -680,34 +662,27 @@ def dispatch(self, request, *args, **kwargs):
         path replacement manually.
         """
 
-        # Manually set the language code to the default language, which isn't prefixed in URLs,
-        # to stop the locale middleware from adding a language/locale prefix back into the study URL
-        # after the request has been dispatched
-        translation.activate(settings.LANGUAGE_CODE)
-        request.LANGUAGE_CODE = settings.LANGUAGE_CODE
-
```

Comment:
I presume that by removing this logic it wasn't solving our problem.  Which is a shame.

Reply:
Yep, doesn't work and I still don't know why. It would've been such a nice solution.